src/CalendarSegment.php

```diff
@@ -388,105 +379,84 @@ public static function showForCalendar(Calendar $calendar)
                 'end'
             ]
         ]);
-        $numrows = count($iterator);
 
+        $daysofweek = Toolbox::getDaysOfWeekArray();
         if ($canedit) {
-            echo "<div class='firstbloc'>";
-            echo "<form name='calendarsegment_form$rand' id='calendarsegment_form$rand' method='post'
-                action='";
-            echo Toolbox::getItemTypeFormURL(__CLASS__) . "'>";
-            echo "<table class='tab_cadre_fixe'>";
-            echo "<tr class='tab_bg_1'><th colspan='7'>" . __('Add a schedule') . "</tr>";
-
-            echo "<tr class='tab_bg_2'><td class='center'>" . _n('Day', 'Days', 1) . "</td><td>";
-            echo "<input type='hidden' name='calendars_id' value='$ID'>";
-            Dropdown::showFromArray('day', Toolbox::getDaysOfWeekArray());
-            echo "</td><td class='center'>" . __('Start') . '</td><td>';
-            Dropdown::showHours("begin", ['value' => date('H') . ":00"]);
-            echo "</td><td class='center'>" . __('End') . '</td><td>';
-            Dropdown::showHours("end", ['value' => (date('H') + 1) . ":00"]);
-            echo "</td><td class='center'>";
-            echo "<input type='submit' name='add' value=\"" . _sx('button', 'Add') . "\" class='btn btn-primary'>";
-            echo "</td></tr>";
-
-            echo "</table>";
-            Html::closeForm();
-            echo "</div>";
+            $twig_params = [
+                'item' => new self(),
+                'calendars_id' => $ID,
+                'days_of_week' => $daysofweek,
+                'begin' => date('H') . ":00",
+                'end' => ((int) date('H') + 1) . ":00",
+                'params' => [
+                    'canedit' => true,
+                ]
+            ];
+            echo TemplateRenderer::getInstance()->renderFromStringTemplate(<<<TWIG
```

Comment:
Please use a dedicated file that will:
1. render this form,
2. include `components/datatable.html.twig` to render the items list.